### DISCUSSION TITLE
Combat, display spell information even for opposing troops

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4689] [Ideas]
During combat it is possible to get information about active spells on the hero's troops, but not on the opponent's troops, because the opponent's creature information window is not permanently displayed with an "EXIT" button.

In fheroes2:

![Spell info combat enemy 001](https://user-images.githubusercontent.com/43583994/143089702-c4aff235-a80a-4df6-ac1d-72f8c2ef4d81.png)

![Spell info combat enemy 002](https://user-images.githubusercontent.com/43583994/143089720-e8c4e107-4c03-40f2-bbad-abcf7a0ee64f.png)

![Spell info combat enemy 003](https://user-images.githubusercontent.com/43583994/143089745-66df3894-b1de-4d50-8c26-9408bf2c6f4b.png)

I suggest to display this same permanent window with an "EXIT" button for the opponent's troops in two ways.
The first one (with the mouse only), by clicking with the middle mouse button on the opponent's troop.
And the second one (with the keyboard and the mouse), by holding down the "ALT" key on the keyboard and right clicking on the opponent's troop.

**Comment by Branikolog** [comment]
> clicking with the middle mouse button on the opponent's troop.

Not everyone owns a mouse with middle button. :)

What if we display the window if we hold right click for quite long time? Let's say 5-6  seconds...

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> > clicking with the middle mouse button on the opponent's troop.
> 
> Not everyone owns a mouse with middle button. :)

That's why I proposed also with the "ALT" key and the right click. :)

> What if we display the window if we hold right click for quite long time? Let's say 5-6 seconds...

I thought about that too, but it has to be quick, even waiting 1 second is too much. :)

**Reply by Branikolog** [reply]
99% of time players use right click for fast check of stats. If player is interested in spell description - I consider this case as "study mode" as in most cases players use this for a few first games. So player is definitely not in hurry. So, waiting like 5 seconds (or even more) seems ok for me in this case....

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> 99% of time players use right click for fast check of stats. If player is interested in spell description - I consider this case as "study mode" as in most cases players use this for a few first games. So player is definitely not in hurry. So, waiting like 5 seconds (or even more) seems ok for me in this case....

I can assure you that this would quickly become annoying, and would be strongly criticized by the players.

**Reply by zenseii** [reply]
What if the window locks if you first press the right mouse button and then while holding it you press the left, so both at the same time?

**Reply by LeHerosInconnu** [reply]
Hello @zenseii,

> What if the window locks if you first press the right mouse button and then while holding it you press the left, so both at the same time?

It could be okay in case the mouse does not have a central button.
But, at the same time, it's annoying (and "risky") because the user has to perform the action in the right order, first the right click held and then the left click.
And if the user makes a wrong manipulation (for example if he doesn't hold the right click correctly) an attack can be engaged without it being what the user wanted to do (and there the computer flies in the room). :)

**Reply by zenseii** [reply]
Good point! That would be really annoying if you mistakingly make a move by left clicking. If it comes down to holding down a button then maybe shift is better since it's already used for this sort of modulation. Alt is also very close to the spacebar.

**Reply by LeHerosInconnu** [reply]
@zenseii.

> Good point! That would be really annoying if you mistakingly make a move by left clicking. If it comes down to holding down a button then maybe shift is better since it's already used for this sort of modulation. Alt is also very close to the spacebar.

As you write, the "SHIFT" key may be preferred to the "ALT" key because of its proximity to the space bar.
Good point too. :)

**Comment by LeHerosInconnu** [comment]
This has been partially implemented, with a left click on the creature's sprite in the “Turn Order” bar.
However, sprites of creatures that are under the influence of certain spells (e.g., blind) are no longer displayed in the “Turn Order” bar and are therefore no longer reachable.
I'll revisit that later.
I close the discussion.
